modules/reindex/src/main/java/org/elasticsearch/index/reindex/RestReindexAction.java

```diff
@@ -68,6 +68,11 @@ protected ReindexRequest buildRequest(RestRequest request) throws IOException {
         if (request.hasParam("scroll")) {
             internal.setScroll(parseTimeValue(request.param("scroll"), "scroll"));
         }
+
+        if (request.hasParam("prefer_v2_templates")) {
+            internal.preferV2Templates(Boolean.parseBoolean(request.param("prefer_v2_templates")));
```

Comment:
for 8.0 we want this to default to `true` if it's not specified by the user (this is just a reminder to change this after backport)

Reply:
I wonder if we should change it here or simply pass null down the road and let create index request pick up correct default (for making it DRY and using only single place to set defaults)